definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/react-redux_v5.x.x.js

```diff
@@ -67,41 +67,45 @@ declare module "react-redux" {
     static +WrappedComponent: WC;
     getWrappedInstance(): React$ElementRef<WC>;
   }
+  // The connection of the Wrapped Component and the Connected Component
+  // happens here in `MP: P`. It means that type wise MP belongs to P,
+  // so to say MP >= P.
   declare type Connector<P, OP, MP: P> = <WC: React$ComponentType<P>>(
     WC,
   ) => Class<ConnectedComponent<OP, WC>> & WC;
 
   // No `mergeProps` argument
 
+  // Got error like inexact OwnProps is incompatible with exact object type?
+  // Just make the OP parameter for `connect()` an exact object.
+  declare type MergeOP<OP, D> = {| ...$Exact<OP>, dispatch: D |};
+  declare type MergeOPSP<OP, SP> = {| ...$Exact<OP>, ...SP |};
+  declare type MergeOPDP<OP, DP> = {| ...$Exact<OP>, ...DP |};
+  declare type MergeOPSPDP<OP, SP, DP> = {| ...$Exact<OP>, ...SP, ...DP |};
```

Comment:
What a name 😂

Reply:
should SP and DP be wrapped to `$Exact` too? Or is it enough that OP is wrapped?

Reply:
To be honest, I really do not know. For the known use cases it's enough. Also, the `OP` is an explicit type parameter while all the others get inferred, and we all know how bad Flow might be in reporting errors if there are multiple layers of type transformations on top. This is why I try to keep it as simple as possible.

Reply:
Ok, makes totally sense! IT can be improved later on if necessary.

Reply:
FWIW, I'm currently working on spread semantics and you won't have to do guesswork about how flow infers the types for spreads anymore.

Reply:
Thanks a lot for making Flow better! Keep up the great job! Can I somehow help you (say, with providing some use cases) or could it be possible to track the progress somewhere to get the typings prepared a bit better this time 😅?

Reply:
Your types as-is should work. I'll write up something more formally when I'm further along.

Basically, spreads for exact objects will stay the same (modulo objects with indexers, that remains to be seen) and spreads with inexact objects will actually work without inferring mixed everywhere. But, there will be more errors when spreading inexact objects.